ShapeFactory.cs

```diff
@@ -0,0 +1,230 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CLIShape
+{
+    internal class ShapeFactory
+    {
+        private Dictionary<char,Shape> shapes = new Dictionary<char, Shape>();
+
+        public ShapeFactory() 
+        {
+
+        }
+
+
+        public Shape createShap(string shape)
+        {
+
+            string input = "";
+            int x, y,z;
+            char id, boolInput;
+            bool fillIn = false;
+            List<string> shapesProp = Shape.Getproperties(shape);
+
+            do
+            {
+                Console.WriteLine($"Would you like to fillIn the {shape}: [Y] for yes [N] for no");
+                boolInput = char.ToUpper(Console.ReadKey().KeyChar);
+                Console.WriteLine();
+
+            } while (!(boolInput == 'Y' || boolInput == 'N'));
+
+            if (boolInput == 'Y')
+                fillIn = true;
+
+            do
+            {
+                Console.WriteLine($"Enter {shape} charId:");
+                id = Console.ReadKey().KeyChar;
+                Console.WriteLine();
+            } while (id==' '||id =='\r' || shapes.Keys.ToList().Contains(id));
+
+
+            do
+            {
+                Console.WriteLine($"Enter {shapesProp[0]} :");
+                input = Console.ReadLine();
+            } while (!int.TryParse(input, out x));
+            
+            if(shapesProp.Count == 1) 
+            {
+                switch (shape)
+                {
+                    case "square":
+                        return new Square(fillIn, id, x);
+                
+                    case "rhombus":
+                        return new Rhombus(fillIn, id, x);
+
+                    case "triangle":
+                        return new Triangle(fillIn, id, x); 
+
+                }
+            }
+
+            do
+            {
+                Console.WriteLine($"Enter {shapesProp[1]} :");
+                input = Console.ReadLine();
+            } while (!int.TryParse(input, out y));
+
+            if(shape == "rectangle") 
+                return new Rectangle(fillIn, id, x, y); 
+
+            do
+            {
+                Console.WriteLine($"Enter {shapesProp[2]} :");
+                input = Console.ReadLine();
+            } while (!int.TryParse(input, out z));
+            
+            return new Trapeze(fillIn, id, x, y, z);    
+
+
+        }
+
+        /// <summary>
+        /// /////////////////
+        /// </summary>
+        /// 
+
+        public void UpdateShap()
+        {
+            char shapeId;
+            Shape shape;
+            char boolInput ;
+
+
+            do
+            {
+                Console.WriteLine($"Enter shap to change charId:");
+                shapeId = Console.ReadKey().KeyChar;
+                Console.WriteLine();
+            } while (shapeId == ' ' || shapeId == '\r' || !shapes.Keys.ToList().Contains(shapeId));
```

Comment:
Consider having a function called `ShapeNotExists` that contains all current and future options.
That way you basically read English:
```
do {
this
} while (ShapeNotExists)
```
A general tip: the more self-explanatory your code is, the better your job is done!